filament/src/ShadowMap.cpp

```diff
@@ -90,27 +90,25 @@ void ShadowMap::render(DriverApi& driver, FView::Range const& range, RenderPass&
     pass.sortCommands();
 }
 
-void ShadowMap::computeSceneCascadeParams(const FScene::LightSoa& lightData, size_t index,
-        FView const& view, filament::CameraInfo const& camera, uint8_t visibleLayers,
-        CascadeParameters& cascadeParams) {
-    // Calculate the directional light's "position".
-    // We could choose any position; we pick the camera position so we have a fixed
-    // reference -- that's "not too far" from the scene.
-    cascadeParams.wsLightPosition = camera.getPosition();
-
-    // Compute the light's model matrix.
-    const float3 lightPosition = cascadeParams.wsLightPosition;
-    const float3 dir = lightData.elementAt<FScene::DIRECTION>(index);
-    const mat4f M = mat4f::lookAt(lightPosition, lightPosition + dir, float3{ 0, 1, 0 });
-    const mat4f Mv = FCamera::rigidTransformInverse(M);
+mat4f ShadowMap::getLightViewMatrix(float3 position, float3 direction) noexcept {
+    const mat4f M = mat4f::lookAt(position, position + direction, float3{ 0, 1, 0 });
+    return FCamera::rigidTransformInverse(M);
+}
+
+void ShadowMap::computeSceneCascadeParams(float3 dir,
+        FScene const& scene, filament::CameraInfo const& camera, uint8_t visibleLayers,
+        SceneInfo& cascadeParams) {
```

Comment:
Rename `cascadeParams` to `sceneInfo`?

Reply:
I realized later that this is both a scene info AND a cascade parameter :-)